File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/Interfaces/DTO/Events/IMessageEventDTO.cs
namespace WorldFeed.Common.Public.Models.Interfaces.DTO.Events
{
    using System;

    using WorldFeed.Common.Public.Models.Enums;

    public interface IMessageEventDTO
    {
        EventType Type { get; set; }

        long Id { get; set; }

        DateTime CreatedAt { get; set; }

        IEventInitiatedViaDTO InitiatedVia { get; set; }

        IMessageCreateDTO MessageCreate { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Interfaces/DTO/Events/IMessageEventDTO.ts
import {InjectionToken} from "@angular/core";

import {EventType} from "../../../Enum/EventType";
import {IEventInitiatedViaDTO} from "../IEventInitiatedViaDTO";
import {IMessageCreateDTO} from "../IMessageCreateDTO";
import {MessageEventDTO} from "../../../../../Core/DTO/Events/MessageEventDTO";
import DateTime from "typescript-dotnet-commonjs/System/Time/DateTime";

export interface IMessageEventDTO {
  type: EventType;
  id: number;
  createdAt: DateTime; // DateTimeOffset;
  initiatedVia: IEventInitiatedViaDTO;
  messageCreate: IMessageCreateDTO;
}

export const IMessageEventDTOToken = new InjectionToken<IMessageEventDTO>('IMessageEventDTO', {
  providedIn: 'root',
  factory: () => new MessageEventDTO(),
});
